+NeuroAnalysis/+IO/Export.m

```diff
@@ -10,39 +10,43 @@
     return;
 end
 %% Get dataset and export path
-result.status = false;
-result.source = '';
 if isa(dataset,'char')
-    result.source = dataset;
+    filename = dataset;
     if ~strcmp(sourceformat,'Unknown')
         dataset = NeuroAnalysis.Base.EvalFun(['NeuroAnalysis.',sourceformat,'.Prepare'],[{dataset},varargin]);
         if ~isempty(dataset) && isfield(dataset,'status') && ~dataset.status
+            result = dataset;
+            result.source = NeuroAnalysis.Base.filenamenodirext(filename);
             return;
         end
         filename = NeuroAnalysis.Base.filenamenodirext(dataset.source);
-        exportpath = fullfile(exportpath,[filename,'.mat']);
+        filepath = fullfile(exportpath,[filename,'.mat']);
     else
         [~,filename,ext]=fileparts(dataset);
-        exportpath=fullfile(exportpath,[filename,ext]);
+        filepath=fullfile(exportpath,[filename,ext]);
     end
 end
 %% Save dataset
-disp(['Exporting Dataset:    ',exportpath,'    ...']);
+disp(['Exporting Dataset:    ',filepath,'    ...']);
 if ~strcmp(sourceformat,'Unknown')
-    dataset.filepath = exportpath;
-    save(exportpath,'dataset','-v7.3');
+    dataset.filepath = filepath;
+    save(filepath,'dataset','-v7.3');
 else
-    copyfile(dataset,exportpath);
+    copyfile(dataset,filepath);
+    dataset = struct;
+    dataset.filepath = filepath;
 end
-result.status = true;
-result.source = filename;
 disp('Exporting Dataset:    Done.');
 %% Callback
 callbackfun = callback{1};
 callbackarg = callback{2};
+callbackresult = struct([]);
 if ~isempty(callbackfun) && ~strcmp(sourceformat,'Unknown')
     disp(['Applying Callback:    ',callbackfun,'    ...']);
-    NeuroAnalysis.Base.EvalFun(callbackfun,[{dataset},callbackarg]);
+    callbackresult = NeuroAnalysis.Base.EvalFun(callbackfun,[{dataset},callbackarg]);
     disp('Applying Callback:    Done.');
 end
-
+%% Update metadata file
+result = NeuroAnalysis.Base.EvalFun('NeuroAnalysis.IO.MetadataExtract', {dataset,exportpath,sourceformat,isparallel,callbackresult});
```

Comment:
i think it's better to separate meta data preparation into individual `NeuroAnalysis.sourceformat.PrepareMetaData` functions. because here is where each dataset have already exported and analyzed, in serial order or parallel. if we just extract aproparate metadata for each dataset and put it in result. we could merge them into the global metadata once after Export whether it is a batch or not. so i don't think we need to worry about `isparallel`, `sourceformat` etc in argument list. just `NeuroAnalysis.Base.EvalFun(['NeuroAnalysis.',sourceformat,'.PrepareMetaData'],[{dataset,callbackresult},varargin])`

Reply:
Ok. We still need a function for updating the metadata file in batch, without exporting. It can be separate, like `NeuroAnalysis.IO.ExportMetadata` or similar, that just opens individual datasets (in parallel or serial) and sends it to the appropriate `PrepareMetadata' function?